src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,193 @@
+                + " ON c.manufacturer_id = m.id WHERE c.id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarByIdStatement
+                        = connection.prepareStatement(getCarByIdQuery)) {
+            getCarByIdStatement.setLong(1, id);
+            ResultSet resultSet = getCarByIdStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCarFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get Car from cars DB by Id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriverListFromResultSet(car.getId()));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> cars = new ArrayList<>();
+        String getAllCarsQuery = "SELECT * FROM cars c JOIN manufacturers m"
+                + " ON c.manufacturer_id = m.id WHERE c.is_deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement
+                        = connection.prepareStatement(getAllCarsQuery)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all Cars from DB", e);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriverListFromResultSet(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarQuery = "UPDATE cars SET (model, manufacturer_id) VALUES (?, ?) "
+                + "WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatement = connection.prepareStatement(updateCarQuery)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update Car " + car, e);
+        }
+        deleteCarsAndDrivers(car);
+        insertCarsAndDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteStatement = "UPDATE cars SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarStatement
+                        = connection.prepareStatement(deleteStatement)) {
+            deleteCarStatement.setLong(1, id);
+            deleteCarsAndDrivers(get(id).get());
```

Comment:
we shouldn't delete data from car_drivers table, because we use soft delete approach